python/semantic_kernel/core_skills/http_skill.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import json
+
+import aiohttp
+
+from semantic_kernel.orchestration.sk_context import SKContext
+from semantic_kernel.skill_definition import sk_function, sk_function_context_parameter
+
+
+class HttpSkill:
+    """
+    A skill that provides HTTP functionality.
+
+    Usage:
+        kernel.import_skill(HttpSkill(), "http")
+
+    Examples:
+
+        {{http.getAsync $url}}
+        {{http.postAsync $url}}
+        {{http.putAsync $url}}
+        {{http.deleteAsync $url}}
+    """
+
+    @sk_function(description="Makes a GET request to a uri", name="getAsync")
+    async def get_async(self, url: str) -> str:
+        """
+        Sends an HTTP GET request to the specified URI and returns
+        the response body as a string.
+        params:
+            uri: The URI to send the request to.
+        returns:
+            The response body as a string.
+        """
+        if not url:
+            raise ValueError("url cannot be `None` or empty")
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, raise_for_status=True) as response:
+                return await response.text()
+
+    @sk_function(description="Makes a POST request to a uri", name="postAsync")
+    @sk_function_context_parameter(name="body", description="The body of the request")
+    async def post_async(self, url: str, context: SKContext) -> str:
+        """
+        Sends an HTTP POST request to the specified URI and returns
+        the response body as a string.
+        params:
+            url: The URI to send the request to.
+            context: Contains the body of the request
+        returns:
+            The response body as a string.
+        """
+
+        has_body, body = context.variables.get("body")
+        if not url:
+            raise ValueError("url cannot be `None` or empty")
+        if not has_body:
+            raise ValueError("no Body found in context")
+        if not body:
```

Comment:
A body isn't required for POSTs or PUTs.

Reply:
thanks updated!

Reply:
Did you push the changes? I'm still seeing errors raised if a body is not found for the post request

Reply:
that is indeed likely that I forgot to push the changes, should be fine now.